### PR TITLE
Build a list of recommendations for consolidated datasets

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Build recommendations according to what has been introduced in https://github.com/opendatateam/udata-recommendations/pull/142

These recommendations will be available at https://schema.data.gouv.fr/api/recommendations.json in the end

**Failing build**: the `api` step will fail until https://github.com/opendatateam/udata/pull/2523 is merged and deployed to production. The current PR relies on the new `schema` facet.

### CODE REVIEW

File: api/recommendations.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import logging
+
+import yaml
+import requests
+import jsonschema
+
+DATA_GOUV_API = "https://www.data.gouv.fr/api/1"
+SCHEMAS_PATH = Path(__file__).parent / "../aggregateur/data/schemas.yml"
+RECOMMENDATION_SCORE = 50
+JSONSCHEMA_URL = "https://raw.githubusercontent.com/AntoineAugusti/udata-recommendations/add-source-score/udata_recommendations/schema.json"
```
Comment (AntoineAugusti) [inline api/recommendations.py:13]: This will need to be updated once the PR is merged

**Comment by AntoineAugusti** [comment]
Just rerun the CircleCI job since the `schema` facet is now available in production

File: .circleci/config.yml [hunk 2]
```diff
@@ -63,6 +63,42 @@ jobs:
           paths:
             - data
             - cache
+  api:
```
Comment (abulte) [inline .circleci/config.yml:66]:
```suggestion
  recommendations:
```

unsure

File: api/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+venv
```
Comment (abulte) [inline api/.gitignore:1]: I usually put those in a global gitignore file on my machines

File: api/recommendations.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+def main():
+    recommendations = []
+    for schema_id, schema_details in consolidated_schemas().items():
+        consolidated_dataset_id = schema_details["consolidation"]["dataset_id"]
+        logger.info(
+            f"Working on schema {schema_id}, consolidated on {consolidated_dataset_id}"
+        )
+        dataset_ids = datasets_for_schema(schema_id)
+        logger.info(f"Found {len(dataset_ids)} associated with schema {schema_id}")
+        for dataset_id in dataset_ids:
```
Comment (abulte) [inline api/recommendations.py:73]: maybe a list comprehension here

**Review by abulte — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (AntoineAugusti) [reply .circleci/config.yml:66]: I don't think it's necessary. I agree we only have recommendations for now but it's likely that there will be multiple scripts in there later, all together building an API

**Pull request merged**